src/OrchardCore.Modules/OrchardCore.Deployment/Services/InMemoryFileBuilder.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Deployment.Services
+{
+    public class InMemoryFileBuilder : IFileBuilder
+    {
+        public byte[] PlanInMemory { get; private set; }
+        public async Task SetFileAsync(string subpath, Stream stream)
+        {
+            //The passed in stream is a MemoryStream which means it is a seekable stream. So the Length always available
+            PlanInMemory = new byte[stream.Length];
+            await stream.ReadAsync(PlanInMemory, 0, (int)stream.Length);
```

Comment:
I think you could just hold the stream here, rather than reading, and then rereading it when you convert to utf8

Probably just return `File(inMemoryFileBuilder.Steam, "application/json")` is enough, and let the controller take reading of streaming it.

Reply:
Tried by saving a reference to the stream into a Stream variable in InMemroyFileBuilder. But a error saying that "Cannot access a closed Stream" when return the result using controller's File().